src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php

```diff
@@ -60,7 +60,7 @@ private function __construct()
      *
      * @throws \InvalidArgumentException if $unprocessedHtml is anything other than a non-empty string
      */
-    public static function fromHtml(string $unprocessedHtml): AbstractHtmlProcessor
+    public static function fromHtml(string $unprocessedHtml): self
```

Comment:
Note: I'd like to keep the `@return static` in the PHPDoc to make it explicit that we return an instance of the corresponding called class.

Reply:
> keep the `@return static` in the PHPDoc to make it explicit

I strongly agree.  There is still not much correlation between PHPDoc and type hints and [probably never will be](https://wiki.php.net/rfc/arrayof).